File: Look and Say (String & Int)/look_and_say.py
def look_and_say(input):
  counter = 1
  answer = ""
  input = str(input)
  for i in range(1, len(input)):
    if input[i] == input[i - 1]:
      counter += 1
    else:
      answer += str(counter)
      answer += input[i-1]
      counter = 1
  last = len(input) - 1
  if input[last - 1] == input[last]:
    answer += str(counter)
    answer += input[last]
  return answer

# Tester code
print "Input: 244466666"
print "Look and say..."
print look_and_say(244466666) # expected: 123456

print "Input: 4443255522"
print "Look and say..."
print look_and_say(4443255522) # expected: 3413123522

File: Look and Say (String & Int)/look_and_say.rb
def look_and_say(input)
  counter = 1
  answer = ""
  input = input.to_s.split("")
  input.each_with_index do |i, index|
    if index > 0
      if i == input[index - 1]
        counter += 1
      else
        answer += counter.to_s + input[index-1]
        counter = 1
      end
    end
  end
  last = input.length - 1
  if input[last - 1] == input[last]
    answer += counter.to_s + input[last]
  end
  answer
end

# Tester code
puts "Input: 244466666"
puts "Look and say..."
puts look_and_say(244466666) # expected: 123456

puts "Input: 4443255522"
puts "Look and say..."
puts look_and_say(4443255522) # expected: 3413123522

puts "Input: 1"
puts "Look and say..."
puts look_and_say(1) # expected: 3413123522